docs/design/plugin_multi_fdb_support.md

```diff
@@ -0,0 +1,134 @@
+The fields are `seedCluster` and `targetClusters`, both contain the following `cluster objects`:
+
+```yaml
+name: sample-cluster-dc3 # Optional: Will overwrite the cluster name in the resulting manifest otherwise uses the metadata
+dcID: dc3 # Must match with the data center ID specified under datacenters
+context: my-kube-ctx 3 # Optional: the kubconfig context to use otherwise the current context will be used
+namespace: dc3 # Optional: Otherwise the namespace from the metadata will be used
+processCounts: # Optional: otherwise the processCounts from he cluster Spec will be used (useful to change cluster specific fields e.g. satellites without SS)
+```
+
+The `targetClusters` will contain a list of all `cluster objects` and the `seedCluster` will contain the `cluster object` to bootstrap the initial
+FDB cluster.
+An example manifest for the `kubectl fdb plugin` could look like this:
+
+```yaml
+apiVersion: apps.foundationdb.org/v1beta1
+kind: FoundationDBCluster
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+    cluster-group: sample-cluster
+  name: sample-cluster
+spec:
+  version: 6.2.29
+  faultDomain:
+    key: foundationdb.org/none
+  processCounts:
+    stateless: -1
+    log: -1
+  databaseConfiguration:
+    redundancy_mode: "double"
+    usable_regions: 2
+    regions:
+      - datacenters:
+          - id: dc1
+            priority: 1
+          - id: dc3
+            satellite: 1
+            priority: 1
+        satellite_logs: 3
+      - datacenters:
+          - id: dc2
+            priority: 0
+          - id: dc3
+            satellite: 1
+            priority: 1
+        satellite_logs: 3
+  seedCluster:
+    name: sample-cluster-dc1
+    dcID: dc1
+    context: my-kube-ctx
+    namespace: dc1
+  targetClusters:
+  - name: sample-cluster-dc2
+    dcID: dc2
+    context: my-kube-ctx
+    namespace: dc2
+  - name: sample-cluster-dc3
+    dcID: dc3
+    context: my-kube-ctx
+    namespace: dc3
+    processCounts:
+      log: 0
+```
+
+The `databaseConfiguration` must be configured how it should be when all FDB clusters are up and running.
+The following field will be automatically set by the `kubectl fdb plugin` when creating the object in Kubernetes:
+
+```yaml
+instanceIDPrefix: $dcID
+dataCenter: $dcID
+seedConnectionString: $connectionString # The seed connection string will be set once the initial cluster is bootstrapped
+```
+
+The Operator will be configured to ignore the fields `seedCluster` and `targetClusters` which means that these fields
+are only used by the `kubectl fdb plugin` and maybe the human operator. Also these fields are optional, if a FDB cluster
+is limited to a single namespace in a single Kubernetes cluster these fields can be omitted.
+
+The `kubectl fdb plugin` will take the following steps when creating a FDB cluster across multiple Kubernetes clusters (or namespaces):
+
+1. Create the seed cluster in the `seedcluster`.
```

Comment:
I think we'll need to change the configuration when creating the seed cluster in a couple ways:

1. If there is region config in the final DC configuration, the region config for the seed cluster should be set to a single data center, containing the DC for the seed cluster, since the other regions won't be available until a later step.
2. If we are using multi-KC replication for the zone, the seed cluster will need to be single-replicated until the other zones are available.